src/Models/ArduinoDeviceBase.ts

```diff
@@ -191,9 +192,23 @@ export abstract class ArduinoDeviceBase implements Device {
                 .replace(rootPathPattern, localAppData.replace(/\\/g, '\\\\'))
                 .replace(versionPattern, this.version);
         await FileUtility.writeFile(type, cppPropertiesFilePath, replaceStr);
+      } else if (plat === 'linux') {
+        const propertiesFilePathLinux =
+            this.extensionContext.asAbsolutePath(path.join(
+                FileNames.resourcesFolderName, FileNames.templatesFolderName,
+                board.id, constants.cppPropertiesFileNameLinux));
+        const propertiesContentLinux =
+            fs.readFileSync(propertiesFilePathLinux).toString();
+        const rootPathPattern = /{ROOTPATH}/g;
+        const versionPattern = /{VERSION}/g;
+        const homeDir = await IoTWorkbenchSettings.getOs();
+        const replaceStr =
+            propertiesContentLinux.replace(rootPathPattern, homeDir)
+                .replace(versionPattern, this.version);
+        await FileUtility.writeFile(type, cppPropertiesFilePath, replaceStr);

```

Comment:
Similar code block as "win32" case, suggest moving to a wrapper method to improve readability.

Reply:
Not all the same since win32 will need changing `\`  to `\\`.